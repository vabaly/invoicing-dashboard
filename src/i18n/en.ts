const en = {
  siteInfo: {
    title: 'Attelas invoicing dashboard',
    description: 'Attelas invoicing dashboard',
  },

  breadcrumbs: {
    home: 'Invoices',
  },

  invoiceTableHead: {
    invoiceNumber: 'Invoice Number',
    clientName: 'Client Name',
    total: 'Total Amount Due',
    dueDate: 'Due Date',
    status: 'Status',
  },

  invoiceDetail: {
    basicInfo: 'Basic Info',
    clientInfo: 'Client Info',
    lineItems: 'Line Items',

    email: 'Email',
    address: 'Address',
    clientId: 'Client ID',
  },

  lineItem: {
    description: 'Description',
    price: 'Price',
    quantity: 'Quantity',
  },

  back: 'Back',
  ok: 'Ok',
  colon: ': ',

  chase: 'Chase',

  email: {
    title: 'Email preview',
    to: 'To',
    subject: 'Subject',
    subjectValue: 'Please check your invoice',
    bodies: {
      0: 'Dear %{name},',
      1: 'This is your invoice details. Please pay it as soon as you can.',
      2: 'Best regards,',
      3: 'Attelas Team',
    },
  },
};

export default en;
