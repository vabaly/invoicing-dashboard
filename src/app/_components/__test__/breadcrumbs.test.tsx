import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import AppBreadcrumbs from '../breadcrumbs';
import i18n from '~/i18n';

jest.mock('next/navigation', () => ({
  usePathname: () => '/path/path segment 1/path segment 2',
}));

describe('AppBreadcrumbs', () => {
  it('should render the breadcrumbs correctly', () => {
    const { asFragment } = render(<AppBreadcrumbs />);

    // Snapshot test
    expect(asFragment().children[0]).toMatchSnapshot();

    // Verify that the breadcrumbs are rendered
    const breadcrumbs = screen.getByRole('navigation');
    expect(breadcrumbs).toBeInTheDocument();

    // Verify that the home breadcrumb is rendered
    const homeBreadcrumb = screen.getByText(i18n.t('breadcrumbs.home'));
    expect(homeBreadcrumb).toBeInTheDocument();

    // Verify that the other breadcrumbs are rendered
    const otherBreadcrumbs = screen.getAllByText(/path segment/);
    expect(otherBreadcrumbs).toHaveLength(2);
  });
});
