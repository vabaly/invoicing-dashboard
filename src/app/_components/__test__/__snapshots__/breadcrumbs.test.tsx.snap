// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBreadcrumbs should render the breadcrumbs correctly 1`] = `
<nav
  aria-label="breadcrumb"
  class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root px-4 css-8yvas4-MuiTypography-root-MuiBreadcrumbs-root"
>
  <ol
    class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
  >
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1fxco1w-MuiTypography-root-MuiLink-root"
        href="/"
      >
        Invoices
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1fxco1w-MuiTypography-root-MuiLink-root"
        href="/path"
      >
        path
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1fxco1w-MuiTypography-root-MuiLink-root"
        href="/path/path segment 1"
      >
        path segment 1
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1f6v1q3-MuiTypography-root"
      >
        path segment 2
      </p>
    </li>
  </ol>
</nav>
`;
